src/Common/SemanticModelExtensions.cs

```diff
@@ -72,10 +72,10 @@ internal static bool IsCallbackOrReturnInvocation(this SemanticModel semanticMod
         };
     }
 
-    private static List<T> GetAllMatchingSymbols<T>(this SemanticModel semanticModel, ExpressionSyntax expression)
+    private static IEnumerable<T> GetAllMatchingSymbols<T>(this SemanticModel semanticModel, ExpressionSyntax expression)
```

Comment:
Return the concrete type (e.g. `ImmutableArray<T>`) instead of the `IEnumerable<T>` interface